Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/CruiseControl.cs

```diff
@@ -0,0 +1,2277 @@
+                        return;
+                    if (!mpc.StateChanged)
+                    {
+                        mpc.StateChanged = true;
+                        mpc.DoMovement(Controllers.MultiPositionController.Movement.Aft);
+                        return;
+                    }
+                }
+            }
+            if (UseThrottleAsSpeedSelector || HasProportionalSpeedSelector)
+                SelectedSpeedDecreasing = true;
+            else
+                SpeedSelectorModeDecrease();
+        }
+        public void SpeedRegulatorSelectedSpeedStopDecrease()
+        {
+            if (Locomotive.MultiPositionControllers != null)
+            {
+                foreach (Controllers.MultiPositionController mpc in Locomotive.MultiPositionControllers)
+                {
+                    if (mpc.controllerBinding != Controllers.MultiPositionController.ControllerBinding.SelectedSpeed)
+                        return;
+                    mpc.StateChanged = false;
+                    mpc.DoMovement(Controllers.MultiPositionController.Movement.Neutral);
+                    return;
+                }
+            }
+            SelectedSpeedDecreasing = false;
+        }
+        public void SpeedRegulatorSelectedSpeedDecrease()
+        {
+            if (!Equipped) return;
+
+            if (selectedSpeedLeverHoldTime + SpeedSelectorStepTimeSeconds > TotalTime)
+                return;
+            selectedSpeedLeverHoldTime = TotalTime;
+            if (SelectedSpeedMpS == 0)
+                return;
+            SelectedSpeedMpS -= SpeedRegulatorNominalSpeedStepMpS;
+            if (SelectedSpeedMpS < 0)
+                SelectedSpeedMpS = 0f;
+            if (MinimumSpeedForCCEffectMpS > 0 &&SelectedSpeedMpS < MinimumSpeedForCCEffectMpS)
+                SelectedSpeedMpS = 0;
+            if (SpeedRegMode == SpeedRegulatorMode.Auto && ForceRegulatorAutoWhenNonZeroSpeedSelected && SelectedSpeedMpS == 0)
+            {
+                // return back to manual, clear all we have controlled before and let the driver to set up new stuff
+                SpeedRegMode = SpeedRegulatorMode.Manual;
+                DynamicBrakePriority = false;
+//                Locomotive.ThrottleController.SetPercent(0);
+//                Locomotive.SetDynamicBrakePercent(0);
+                Locomotive.DynamicBrakeChangeActiveState(false);
+            }
+            if (SelectedSpeedMpS == 0)
+            {
+                if (HasProportionalSpeedSelector)
+                {
+                    Locomotive.SignalEvent(Common.Event.LeverToZero);
+                }
+                SelectedSpeedDecreasing = false;
+            }
+            if (SpeedIsMph)
+                Simulator.Confirmer.Message(ConfirmLevel.Information, Simulator.Catalog.GetString("Selected speed changed to ") + Math.Round(MpS.FromMpS(SelectedSpeedMpS, false), 0, MidpointRounding.AwayFromZero).ToString() + " mph");
```

Comment:
Better localization

Reply:
What do you mean here? Localizing "mph"?

Reply:
Done